pinot-core/src/main/java/org/apache/pinot/core/util/QueryOptionsUtils.java

```diff
@@ -102,4 +102,8 @@ public static boolean isNullHandlingEnabled(Map<String, String> queryOptions) {
   public static boolean isServerReturnFinalResult(Map<String, String> queryOptions) {
     return Boolean.parseBoolean(queryOptions.get(QueryOptionKey.SERVER_RETURN_FINAL_RESULT));
   }
+
+  public static boolean isSkipOrderByOptimization(Map<String, String> queryOptions) {
+    return Boolean.parseBoolean(queryOptions.getOrDefault("skip-order-by", "false"));
```

Comment:
I've replaced it with a new enum option `order-by-algorithm`. The only literal right now is to apply the `naive` algorithm, which means to run with the brute force algorithm, but we could add more literals in the future to be treated as a hint or as a mandatory algorithm (which is very useful in benchmarks, but may be useful in case there is a bug in some algorithm).